src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -1815,22 +1815,33 @@ class GroupByClauseSegment(ansi.GroupByClauseSegment):
         "GROUP",
         "BY",
         Indent,
-        Delimited(
-            OneOf(
-                Ref("ColumnReferenceSegment"),
-                # Can `GROUP BY 1`
-                Ref("NumericLiteralSegment"),
-                # Can `GROUP BY coalesce(col, 1)`
-                Ref("ExpressionSegment"),
-                Ref("CubeRollupClauseSegment"),
-                Ref("GroupingSetsClauseSegment"),
+        OneOf(
+            Delimited(
+                AnyNumberOf(
```

Comment:
You don't need this AnyNumberOf - as Delimited extends AnyOf, and the Delimited isn't optional